src/components/settings/SettingsAccessTab/SettingsAccessGenerateLinkButton.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useContext } from 'react';
+import PropTypes from 'prop-types';
+import {
+  StatefulButton,
+} from '@edx/paragon';
+import moment from 'moment';
+import { logError } from '@edx/frontend-platform/logging';
+
+import LmsApiService from '../../../data/services/LmsApiService';
+import { SettingsContext } from '../SettingsContext';
+
+const BUTTON_PROPS = {
+  labels: {
+    default: 'Generate link',
+    pending: 'Generating link',
```

Comment:
I'm ok with this, assuming we are ok that users will be presented with a disabled button (for not a very long time) with no real feedback as to why it's disabled.

Reply:
@manny-m Like you mentioned, because it's going to be disabled for such a short period of time (should be hardly noticeable on stage/production), disabling the button briefly feels preferable to showing a flash of long button text that the user likely won't have time to read anyways.

If we absolutely need to optimize for showing any feedback about loading state, we could do it conditionally. Disable the button for the first 200ms and then conditionally show the longer text for any duration of loading time longer than that.

Another reasonable option: don't render the "Generate link" button until the customer agreement data is loaded (avoids the problem of disabled button without feedback).